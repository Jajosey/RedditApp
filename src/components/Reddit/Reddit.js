import React from "react";
import styled from "styled-components";
import Votes from "../Votes/Votes";

const RedditDiv = styled.div`
  margin: 5px;
  padding: 10px;
  font-family: arial, sans-serif;
  font-size: 15px;
  font-weight: bold;
  border: 1px solid blue;
  text-align: center;
  box-shadow: 10px 10px 5px grey;
  &:hover {
    background-color: #00ffff;
  }
`;

const WrapperDiv = styled.div`
  display: grid;
  margin: 10px;

  grid-template-columns: 25% 25% 25% 25%;

  @media (max-width: 800px) {
    flex-direction: column;
    grid-template-columns: 50% 50%;
  }
`;

const AuthorDiv = styled.div`
  font-size: 10px;
  color: gray;
`;

const CommentsDiv = styled.div`
  background: ${object => (object.score % 2 === 0 ? "orange" : "blue")};
`;
const dataToLinkComponent = (permalink, title) => {
  const url = `https://www.reddit.com/${permalink}`;
  return <a href={url}>{title}</a>;
};

const Reddit = props => {
  const resource = props.redditData;
  return (
    <WrapperDiv>
      {resource.map((object, index) => (
        <RedditDiv key={index}>
          <div>
            {dataToLinkComponent(object.permalink, object.title)}
            <AuthorDiv>By: {object.author}</AuthorDiv>
            <CommentsDiv {...object}>
              {object.num_comments} Comments
            </CommentsDiv>
            <Votes ups={object.ups} downs={object.downs} />
          </div>
        </RedditDiv>
      ))}
    </WrapperDiv>
  );
};

export default Reddit;
