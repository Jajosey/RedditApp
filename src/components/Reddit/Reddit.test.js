import React from 'react';
import renderer from 'react-test-renderer'
import Reddit from './Reddit'

const MOCK_RESPONSE = [
    
        {
          kind: "t3",
          data: {
            author: "TestAuthor111",
            title: "An Interesting Title",
            num_comments: 54,
            permalink: "/r/ProgrammerHumor/comments/fa7b7f/an_interesting_title/",
            ups: 231,
            downs: 0,
            score: 231
          }
        },
        {
          kind: "t3",
          data: {
            author: "TestAuthor222",
            title: "Another Interesting Title",
            num_comments: 22,
            permalink:
              "/r/ProgrammerHumor/comments/uu2u2u/another_interesting_title/",
            ups: 222,
            downs: 0,
            score: 222
          }
        }
      ]
  

describe('<Reddit/>', () => {
    it('matches the snapshot', () =>{
        const tree = renderer.create(<Reddit redditData={MOCK_RESPONSE}/>).toJSON()
        expect(tree).toMatchSnapshot();
    })
})