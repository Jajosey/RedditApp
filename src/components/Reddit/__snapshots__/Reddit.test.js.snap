// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Reddit/> matches the snapshot 1`] = `
<div
  className="sc-AxiKw hAyeDl"
>
  <div
    className="sc-AxirZ hphjTJ"
  >
    <div>
      <a
        href="https://www.reddit.com/undefined"
      />
      <div
        className="sc-AxhCb iGBEki"
      >
        By: 
      </div>
      <div
        className="sc-AxhUy jJQdxR"
        data={
          Object {
            "author": "TestAuthor111",
            "downs": 0,
            "num_comments": 54,
            "permalink": "/r/ProgrammerHumor/comments/fa7b7f/an_interesting_title/",
            "score": 231,
            "title": "An Interesting Title",
            "ups": 231,
          }
        }
        kind="t3"
      >
         Comments
      </div>
      <div>
        <div
          className="sc-AxjAm ecwSUW"
        >
          Up Votes:
           
        </div>
        <div
          className="sc-AxjAm ecwSUW"
        >
          Down Votes:
           
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-AxirZ hphjTJ"
  >
    <div>
      <a
        href="https://www.reddit.com/undefined"
      />
      <div
        className="sc-AxhCb iGBEki"
      >
        By: 
      </div>
      <div
        className="sc-AxhUy jJQdxR"
        data={
          Object {
            "author": "TestAuthor222",
            "downs": 0,
            "num_comments": 22,
            "permalink": "/r/ProgrammerHumor/comments/uu2u2u/another_interesting_title/",
            "score": 222,
            "title": "Another Interesting Title",
            "ups": 222,
          }
        }
        kind="t3"
      >
         Comments
      </div>
      <div>
        <div
          className="sc-AxjAm ecwSUW"
        >
          Up Votes:
           
        </div>
        <div
          className="sc-AxjAm ecwSUW"
        >
          Down Votes:
           
        </div>
      </div>
    </div>
  </div>
</div>
`;
